### ISSUE TITLE
Fix Request: false positive in AvoidInMemoryStreamingDefaultConstructor for Kotlin

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#616]
initialSize is provided:
```kotlin
private fun withPiWriter(onlyForMuzzlPMD: Int, proc: Consumer<PiWriter>): String {
        StringWriter(onlyForMuzzlPMD).use { stringWriter ->
            PiWriter(stringWriter).use { piWriter ->
                proc.accept(piWriter)
                piWriter.flush()
                return stringWriter.toString()
            }
        }
    }
```

**Comment by jborgers** [comment]
other example:
```kotlin
fun toGzip(decompressed: ByteArray) =
        runCatching {
            ByteArrayOutputStream(decompressed.size).use { baos ->
                GZIPOutputStream(baos).use { gzos ->
                    gzos.write(decompressed)
                    gzos.finish()
                    baos.toByteArray()
                }
            }
        }
}
```

**Comment by jborgers** [comment]
yet another:
```kotlin
private fun readBodyBytes(dbf: List<DataBuffer>): ByteArray {
        return ByteArrayOutputStream(BUFFER_SIZE_BYTES).use { boas ->
            dbf.forEach { db ->
                val b = db.asInputStream().readBytes()
                boas.writeBytes(b)
            }
            boas.toByteArray()
        }
}
```
